plugins/inputs/jolokia2/jolokia.go

```diff
@@ -0,0 +1,169 @@
+package jolokia2
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Jolokia struct {
+	Agents                agentsConfig
+	Proxy                 proxyConfig
+	Metrics               []metricConfig `toml:"metric"`
+	DefaultFieldPrefix    string         `toml:"default_field_prefix"`
+	DefaultFieldSeparator string         `toml:"default_field_separator"`
+	DefaultTagPrefix      string         `toml:"default_tag_prefix"`
+	DefaultTagSeparator   string         `toml:"default_tag_separator"`
+}
+
+type remoteConfig struct {
+	ResponseTimeout    time.Duration `toml:"response_timeout"`
+	Username           string
+	Password           string
+	SSLCA              string `toml:"ssl_ca"`
+	SSLCert            string `toml:"ssl_cert"`
+	SSLKey             string `toml:"ssl_key"`
+	InsecureSkipVerify bool   `toml:"insecure_skip_verify"`
+}
+
+type agentsConfig struct {
+	remoteConfig
+	Urls []string
+}
+
+type proxyConfig struct {
+	remoteConfig
+	Url                   string
+	DefaultTargetPassword string `toml:"default_target_username"`
+	DefaultTargetUsername string `toml:"default_target_password"`
+
+	Targets []proxyTargetConfig
+}
+
+type proxyTargetConfig struct {
+	Url      string
+	Username string
+	Password string
+}
+
+type metricConfig struct {
+	Name           string
+	Mbean          string
+	Paths          []string
+	FieldName      *string  `toml:"field_name"`
+	FieldPrefix    *string  `toml:"field_prefix"`
+	FieldSeparator *string  `toml:"field_separator"`
+	TagPrefix      *string  `toml:"tag_prefix"`
+	TagSeparator   *string  `toml:"tag_separator"`
+	TagKeys        []string `toml:"tag_keys"`
+	UntagKeys      []string `toml:"untag_keys"`
+}
+
+func (jc *Jolokia) SampleConfig() string {
+	return fmt.Sprintf(`
+# %s
```

Comment:
you don't need to put the Description in your SampleConfig, telegraf will add that automatically